Replace rustc_serialize with serde_json in style_tests

mhaessig opened the pull request:
#12410 Stop using rustc_serialize

Replaced rustc_serialize with serde_json in
- [ ] ~~components/config/Cargo.toml~~
- [ ] ~~components/config/lib.rs~~
- [ ] ~~components/config/prefs.rs~~
- [ ] ~~components/script_traits/Cargo.toml~~
- [ ] ~~components/script_traits/lib.rs~~
- [ ] ~~components/script_traits/webdriver_msg.rs~~
- [ ] ~~components/webdriver_server/Cargo.toml~~
- [ ] ~~components/webdriver_server/lib.rs~~
- [X] tests/unit/style/Cargo.toml
- [X] test/unit/style/lib.rs
- [X] test/unit/style/properties/scaffolding.rs

PR checks:
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [X] These changes fix (partially) #12410.
- [X] These changes do not require tests because functionality was not changed or a test itself was edited

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/18710)
<!-- Reviewable:end -->

mhaessig committed:
removed rustc_serialize in style tests

KiChjang commented on tests/unit/style/properties/scaffolding.rs:
```diff
@@ -51,8 +52,9 @@ fn find_python() -> String {
+        } else {
+            "python"
+        }
+        .to_owned()
```
What's the deal with this indentation change?

mhaessig commented:
Sorry, forgot to turn off auto-format in the editor. I updated the commit with the correct indentation.

bors-servo commented:
:umbrella: The latest upstream changes (presumably #18728) made this pull request unmergeable. Please resolve the merge conflicts.

jdm commented:
@mhaessig I'm assuming that WIP means that you're planning to continue working on this PR before it gets merged?

mhaessig commented:
@jdm Yes, I do.

mhaessig commented:
I rebased the commit and removed the conflict. I hope it's good to merge now.

This issue turned out to be harder then I taught in the beginning. All the stuff related to webdriver_server is too advanced for me at this time. 
I will update the issue with the list of files that still have rustc_serialize in them and ask to be unassigned from the issue.

mhaessig commented:
@KiChjang Would you mind reviewing rebase on this PR?

KiChjang commented:
Do you mean you've rebased and you want to ensure that everything looks fine so far? In that case, the answer is yes.

mhaessig commented:
Actually I would like to merge this, because the code related to `webdriver_server` is too advanced for me at this point. Because I don't know when I can get back to it I'd like to free up the issue so someone else could give it a shot.

jdm commented:
@bors-servo: r+
Sorry about the delay!

bors-servo commented:
:pushpin: Commit 93cee84 has been approved by `jdm`

<!-- @bors-servo r=jdm 93cee84f9ffc1a947fd756e94fd322d42d0bbf3a -->

dtolnay commented on tests/unit/style/properties/scaffolding.rs:
```diff
@@ -25,10 +25,11 @@ fn properties_list_json() {
     assert!(status.success());
-    let properties = Json::from_reader(&mut File::open(json).unwrap()).unwrap();
+
+    let properties: Value = serde_json::from_reader(File::open(json).unwrap()).unwrap();
```
You could deserialize directly to [`serde_json::Map<String, Value>`](https://docs.serde.rs/serde_json/struct.Map.html) instead of `Value` and save those `as_object().unwrap()` calls.

bors-servo commented:
:hourglass: Testing commit 93cee84f9ffc1a947fd756e94fd322d42d0bbf3a with merge 2977a4e26e0691dad8b640b47edbf77968d97d5a...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/9786), [arm32](http://build.servo.org/builders/arm32/builds/10020), [arm64](http://build.servo.org/builders/arm64/builds/10050), [linux-dev](http://build.servo.org/builders/linux-dev/builds/10068), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/6898), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/6924), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/10126), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/2560), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/2576), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/6821), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/6887), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/2639), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/2601), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/5443)
Approved by: jdm
Pushing 2977a4e26e0691dad8b640b47edbf77968d97d5a to master...
